### ISSUE TITLE
Compiling latest master branch error due to characters depreciated Xcode 9.1

### ISSUE THREAD

**Issue by darren102** [realm/SwiftLint#1938]
### Bug Report

```bash
/ArchiveIntermediates/swiftlint/IntermediateBuildFilesPath/sourcekitten.build/Release/SourceKittenFramework.build/Objects-normal/x86_64/SwiftLangSyntax.o
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/SourceDeclaration.swift:120:60: error: 'characters' is deprecated: Please use String or Substring directly
        let setterOffset = accessorType.propertyTypeString.characters.count
                                                           ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:176:46: error: 'characters' is deprecated: Please use String or Substring directly
                    character = line.content.characters.prefix(prefixLength).reduce(0) { sum, character in
                                             ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:176:46: error: 'characters' is deprecated: Please use String or Substring directly
                    character = line.content.characters.prefix(prefixLength).reduce(0) { sum, character in
                                             ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:418:23: error: 'characters' is deprecated: Please use String or Substring directly
        return String(characters.prefix(1)).capitalized + String(characters.dropFirst())
                      ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:418:66: error: 'characters' is deprecated: Please use String or Substring directly
        return String(characters.prefix(1)).capitalized + String(characters.dropFirst())
                                                                 ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:557:94: error: 'characters' is deprecated: Please use String or Substring directly
            if lineLeadingCharacters < minLeadingCharacters && lineLeadingWhitespace != line.characters.count {
                                                                                             ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:563:21: error: 'characters' is deprecated: Please use String or Substring directly
            if line.characters.count >= minLeadingCharacters {
                    ^
/Users/darren/src/SwiftLint/Carthage/Checkouts/SourceKitten/Source/SourceKittenFramework/String+SourceKitten.swift:563:21: error: 'characters' is deprecated: Please use String or Substring directly
            if line.characters.count >= minLeadingCharacters {
                    ^

** INSTALL FAILED **


The following build commands failed:
	CompileSwift normal x86_64
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(2 failures)
make: *** [installables] Error 65
machost:SwiftLint darren$ 

```
### Environment

* Trying to build from source

**Comment by SDGGiesbrecht** [comment]
@darren102,

Those errors are in SourceKitten, not SwiftLint (Though there are likely similar deprecations used in SwiftLint and its other dependencies as well.) It will take time for the changes to filter up the dependency chain.

In the meantime:

Those are normally only warnings. Somewhere you have set warnings to be elevated to errors. If you let warnings be, SwiftLint will still compile successfully.

Alternatively you can use the [Swift Version Manager](https://github.com/kylef/swiftenv) or some other method to temporarily switch back to an earlier version of Swift. The warnings you posted are new to Swift 4.0.2.

**Comment by darren102** [comment]
@SDGGiesbrecht i'll check the warnings to errors piece, however i just did a clean clone and did the make install and received this. Is there some config i'm missing to turn off the warnings as errors?

**Comment by SDGGiesbrecht** [comment]
@darren102, that can be controlled by Xcode preferences (which is in your hands) or flags to `swift`, `xcodebuild`, etc, which would be directly or indirectly buried somewhere in the Makefile.

Since SwiftLint supports the Swift Package Manager, you can get around both by simply running `swift build --configuration release` and then doing something with the executable that appears in `.build/release`.

If you just want to play around without a permanent install, `swift run swiftlint ...` will also do the trick (from the repository root).

**Comment by darren102** [comment]
Thanks @SDGGiesbrecht building the release config did the trick

**Comment by SDGGiesbrecht** [comment]
All such warnings were likely resolved by #1943.

**Comment by AdrianBinDC** [comment]
If you encounter this issue, you can go into your target, click "Build Settings", and search for "Treat Warnings as Errors". Ensure the value is set to "No".
